pinot-core/src/main/java/org/apache/pinot/core/transport/server/routing/stats/ServerRoutingStatsManager.java

```diff
@@ -377,4 +389,15 @@ public Double fetchHybridScoreForServer(String server) {
       stats.getServerReadLock().unlock();
     }
   }
+
+  private void recordQueueSizeMetrics() {
+    int queueSize = getQueueSize();
+    _brokerMetrics.setValueOfGlobalGauge(BrokerGauge.ROUTING_STATS_MANAGER_QUEUE_SIZE, queueSize);
+    if (queueSize > _executorQueueSizeWarnThreshold) {
+      _brokerMetrics.addMeteredGlobalValue(BrokerMeter.ROUTING_STATS_MANAGER_Q_LIMIT_REACHED, 1L);
+      LOGGER.warn(String.format("Stats Manager queue size exceeds warn threshold = %d. "
+              + "Current queue size = %d, completed task count = %d.",
+          _executorQueueSizeWarnThreshold, queueSize, getCompletedTaskCount()));
```

Comment:
This can lead to a lot of logs during high-qps use-cases, isn't it?

Reply:
It could. For high qps, we can increase the warn threshold as needed or disable the logging if it can cause issues. 

This diff mainly focus on having the visibility on stats updates.

Reply:
Good call. I think we should remove this log line. We have metrics already for this and stats manager queue size being high is not really a major issue.

Reply:
got it, removed the logging the new commit